api/src/services/users.js

```diff
@@ -432,51 +480,218 @@ const missingFields = data => {
 
 const getUpdatedUserDoc = (username, data) => {
   const userID = createID(username);
-  return module.exports._validateUser(userID).then(doc => {
-    const user = Object.assign(doc, module.exports._getUserUpdates(username, data));
+  return validateUser(userID).then(doc => {
+    const user = Object.assign(doc, getUserUpdates(username, data));
     user._id = userID;
     return user;
   });
 };
 
 const getUpdatedSettingsDoc = (username, data) => {
   const userID = createID(username);
-  return module.exports._validateUserSettings(userID).then(doc => {
-    const settings = Object.assign(doc, module.exports._getSettingsUpdates(username, data));
+  return validateUserSettings(userID).then(doc => {
+    const settings = Object.assign(doc, getSettingsUpdates(username, data));
     settings._id = userID;
     return settings;
   });
 };
 
+const validTokenLoginConfig = () => {
+  const tokenLoginConfig = config.get('token_login');
+  const isValidConfig = tokenLoginConfig &&
+                        tokenLoginConfig.enabled &&
+                        (tokenLoginConfig.translation_key || tokenLoginConfig.message) &&
+                        tokenLoginConfig.app_url;
+  return isValidConfig;
+};
+
+const shouldEnableTokenLogin = data => validTokenLoginConfig() && data.token_login;
+const shouldDisableTokenLogin = data => data.token_login === false;
+
+// this is not meant to be secure, just consistent!
+const encodeUsername = username => {
+  return Buffer.from(username).toString('base64');
```

Comment:
Any other encryption mechanism generated absurdly long string when encrypting, and I didn't want to store this into an additional doc - I don't want authentication to depend on a document in medic db.

Reply:
Why encode it at all? Or consider using `encodeURIComponent`?

Reply:
I feel uncomfortable sending the username in clear text.